File: code/Cracking/HardProblem/SurveillanceSystem/solve/cpp/SurveillanceSystemRunner.cpp
#include <topcoder>
using namespace TopCoder;
#include "SurveillanceSystem.cpp"
#include <fstream>
using namespace std;

int main(int argc, char const *argv[]) {
    try {
        ifstream ifs(argv[1]);
        string containers; read(ifs, containers); next(ifs);
        vector<int> reports; read(ifs, reports); next(ifs);
        int L; read(ifs, L);
        ifs.close();
        cout<<argv[1]<<endl;
        ofstream ofs(argv[2]);
        show(ofs, getContainerInfo(containers, reports, L));
        ofs.close();
    } catch (exception &e) {
        cerr << e.what() << endl;
    }
    return 0;
}

File: code/Cracking/HardProblem/SurveillanceSystem/solve/java/SurveillanceSystemRunner.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class SurveillanceSystemRunner {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        String containers = (String) reader.next(String.class);
        reader.next();
        
        List<Integer> reportsBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] reports = new int[reportsBoxed.size()];
        for (int _i = 0; _i < reportsBoxed.size(); ++_i)
            reports[_i] = reportsBoxed.get(_i);
        reader.next();
        
        int L = (Integer) reader.next(Integer.class);
        reader.close();

        SurveillanceSystem solver = new SurveillanceSystem();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.getContainerInfo(containers, reports, L));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
